OpenRA.Game/Map/Map.cs

```diff
@@ -760,9 +802,8 @@ public void FixOpenAreas(Ruleset rules)
 
 		public byte GetTerrainIndex(CPos cell)
 		{
-			var uv = cell.ToMPos(this);
-			var custom = CustomTerrain[uv];
-			return custom != byte.MaxValue ? custom : cachedTileSet.Value.GetTerrainIndex(MapTiles.Value[uv]);
+			var custom = CustomTerrain[cell];
+			return custom != byte.MaxValue ? custom : cachedTileSet.Value.GetTerrainIndex(MapTiles.Value[cell]);
```

Comment:
Weirdly, the conversation from CPos to MPos is expensive accordning to profiling, otherwise I would have done that. This change is intended to avoid the CPos->MPos conversion if possible.